provider/provider.go

```diff
@@ -279,6 +279,12 @@ func configureDatabricksClient(ctx context.Context, d *schema.ResourceData) (any
 			cfg.AuthType = newer
 		}
 	}
+	cfg.EnsureResolved()
```

Comment:
do we still have the lazy authentication here?

Reply:
Yeah. `EnsureResolved` is actually called in `client.New()`. It makes sure that we've captured settings from the users .databrickscfg profile and/or environment variables, but it doesn't authenticate at all.